sdk/dotnet/Pulumi.Automation/LocalWorkspace.cs

```diff
@@ -518,14 +518,14 @@ public override async Task<ConfigValue> GetConfigAsync(string stackName, string
         }
 
         /// <inheritdoc/>
-        public override async Task SetConfigAsync(string stackName, string key, ConfigValue value, CancellationToken cancellationToken = default)
+        public override Task SetConfigAsync(string stackName, string key, ConfigValue value, CancellationToken cancellationToken = default)
         {
             var secretArg = value.IsSecret ? "--secret" : "--plaintext";
-            await this.RunCommandAsync(new[] { "config", "set", key, value.Value, secretArg, "--stack", stackName }, cancellationToken).ConfigureAwait(false);
+            return this.RunCommandAsync(new[] { "config", "set", key, value.Value, secretArg, "--stack", stackName }, cancellationToken);
```

Comment:
Is this done by resharper? Two versions aren't exactly the same, there are some subtle difference in exception propagation, but I think it shouldn't matter for us here. Just interesting if the tool is confident to suggest this.

Reply:
No, that's not coming from Resharper. I installed the latest versions of the analyzers suggested [here](https://cezarypiatek.github.io/post/async-analyzers-summary/):
```xml
<ItemGroup>
  <PackageReference Include="AsyncFixer" Version="1.5.1">
    <PrivateAssets>all</PrivateAssets>
    <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
  </PackageReference>
  <PackageReference Include="Asyncify" Version="0.9.7" >
    <PrivateAssets>all</PrivateAssets>
    <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
  </PackageReference>
  <PackageReference Include="Meziantou.Analyzer" Version="1.0.667">
    <PrivateAssets>all</PrivateAssets>
    <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
  </PackageReference>
  <PackageReference Include="Microsoft.VisualStudio.Threading.Analyzers" Version="16.10.56" >
    <PrivateAssets>all</PrivateAssets>
    <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
  </PackageReference>
</ItemGroup>
```
This particular warning is "AsyncFixer01: The method `SetConfigAsync` does not need to use async/await."

Reply:
Normally there's no need to await unless it's a synchronization point after which something else needs to be executed. In this case it's a tail call in the function and clearly nothing follows it at that scope, so awaiting can be deferred to an outer scope.

Reply:
Capturing the caveats:
* [Redundant async/await](https://cezarypiatek.github.io/post/async-analyzers-p1/#1-redundant-asyncawait)
* [Eliding Async and Await](https://blog.stephencleary.com/2016/12/eliding-async-await.html)

TL;DR: if in doubt we can leave them in to avoid possible subtle changes in behavior.

Reply:
@t0yv0 re tooling see the analyzers above. I'm not sure I'd recommend integrating them more permanently though as they spew other suggestions / warnings which I was ignoring to focus on async await.